### PR TITLE
Product Price: Transition from using CSS margin to Global Styles

### PR DESCRIPTION
**Pull request by danieldudzic**
As a part of https://github.com/woocommerce/woocommerce-blocks/issues/7954 I'm porting product elements to utilize Global Styles instead of CSS.

This PR moves the margin from CSS to Global Styles for the **Product Price** element. 

#### User Facing Testing

1. Add a **Products** block on a new page.
2. Ensure the **Product Price** `margin` is displaying correctly (and is present in the Dimensions > Margin setting).
3. Iterate over each **Products** block pattern. Make sure the `margin` is correct.

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Product Price: Move margin from CSS to Global Styles.

### CODE REVIEW

File: assets/js/base/components/product-price/index.tsx [hunk 1]
```diff
@@ -239,6 +239,10 @@ export interface ProductPriceProps {
 	 * Custom style to apply to the regular price wrapper.
 	 */
 	regularPriceStyle?: React.CSSProperties | undefined;
+	/**
+	 * Custom margin to apply to the price wrapper.
+	 */
+	spacingStyle?: React.CSSProperties | undefined;
```
Comment (sunyatasattva) [inline assets/js/base/components/product-price/index.tsx]: `React.CSSProperties` includes all possible CSS Properties. While `spacingStyle` should include only spacing related ones, right? Like `margin` and `padding` only? Or is it only `margin`?

File: assets/js/blocks/product-query/constants.ts [hunk 2]
```diff
@@ -90,7 +90,15 @@ export const INNER_BLOCKS_TEMPLATE: InnerBlockTemplate[] = [
 			],
 			[
 				'woocommerce/product-price',
-				{ textAlign: 'center', fontSize: 'small' },
+				{
+					textAlign: 'center',
+					fontSize: 'small',
+					style: {
+						spacing: {
+							margin: { bottom: 'var:preset|spacing|30' },
```
Comment (sunyatasattva) [inline assets/js/blocks/product-query/constants.ts]: Just so I learn more about this API, how does this work?

**Review by sunyatasattva — CHANGES_REQUESTED** [review]
Only TS change. Otherwise, code looks good!

File: assets/js/base/components/product-price/index.tsx (hunk 1 above)
Comment (danieldudzic) [reply assets/js/base/components/product-price/index.tsx]: Right. For now it's just `margin`.

File: assets/js/blocks/product-query/constants.ts (hunk 2 above)
Comment (danieldudzic) [reply assets/js/blocks/product-query/constants.ts]: It pre-sets the bottom margin to `1`:
Edit_Page_“Product_Query”_‹_ratings_—_WordPress

**Comment by danieldudzic** [comment]
I have re-added CSS margins for the **All Products** block context, so as to not visually break it for existing instances of the **All Products** block.

**Comment by danieldudzic** [comment]
Failing tests seem to be unrelated to the committed code. I will go ahead and merge this.

**Pull request merged**